pandas/tseries/base.py

```diff
@@ -330,11 +329,16 @@ def _nat_new(self, box=True):
     def map(self, f):
         try:
             result = f(self)
-            if not isinstance(result, (np.ndarray, Index)):
+
+            # Try to use this result if we can
+            if isinstance(result, np.ndarray):
+                self._shallow_copy(result)
+
+            if not isinstance(result, Index):
                 raise TypeError
```

Comment:
I *think* this is caught in an outer scope and so is not seen. but can you add an informative message to this ``TypeError``